File: integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/DuplicateRepository.java
package io.quarkus.it.panache.reactive;

import jakarta.enterprise.context.ApplicationScoped;

import io.quarkus.hibernate.reactive.panache.PanacheRepositoryBase;
import io.smallrye.mutiny.Uni;

@ApplicationScoped
public class DuplicateRepository implements PanacheRepositoryBase<DuplicateEntity, Integer> {

    @Override
    public Uni<DuplicateEntity> findById(Integer id) {
        DuplicateEntity duplicate = new DuplicateEntity();
        duplicate.id = id;
        return Uni.createFrom().item(duplicate);
    }
}

File: integration-tests/hibernate-reactive-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/reactive/kotlin/DuplicateRepository.kt
package io.quarkus.it.panache.reactive.kotlin

import io.quarkus.hibernate.reactive.panache.kotlin.PanacheRepositoryBase
import io.smallrye.mutiny.Uni

class DuplicateRepository : PanacheRepositoryBase<DuplicateEntity, Int> {
    override fun findById(id: Int): Uni<DuplicateEntity> {
        return super.findById(id)
    }
}